### PR TITLE
cpu/esp32: replace RTC implementation by RTT implementation

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR replaces the RTC implementation with an RTT implementation that uses the RTC hardware timer of the ESP32. The RTC implementation used instead is the RTT based RTC implementation provided by the `rtt_rtc` module.

Although the RTC hardware module of the ESP32 is called an RTC, it is a Real Time Timer (RTT) rather than a Real Timer Clock. Therefore, the former RTC peripheral driver implementation is replaced by an RTT periphal driver implementation.

**Note:** An RTT/RTC alarm is not preserved during reboot or deep-sleep.

### Testing procedure
`tests/periph_rtc` and `tests/periph_rtc` should work
```
make BOARD=esp32-wroom-32 -C tests/periph_rtt flash test
```
```
make BOARD=esp32-wroom-32 -C tests/periph_rtc flash test
```
Furthermore, use `tests/periph_pm` can be used to test that the RTT and RTC are retained in sleep modes and during reboot.
```
CFLAGS='-DESP_PM_WUP_PINS=GPIO0 -DESP_PM_WUP_LEVEL=ESP_EXT1_WAKEUP_ALL_LOW' \
USEMODULE='shell_commands rtt_cmd' \
make BOARD=esp32-wroom-32 -C tests/periph_pm flash term
```
```
> rtc settime 2020-03-29 16:17:40
> rtc gettime
2020-03-29 16:17:43
> reboot
...
> rtc gettime
2020-03-29 16:17:49
> set 0
CPU will enter power mode 0.
```
Press `BTN0`
```
...
> rtc gettime
2020-03-29 16:18:01
```

### CODE REVIEW

**Comment by benpicco** [comment]
Could it be that the toolchain on Murdock was upgraded and now `compile/tests/libc_newlib/esp32-wrover-kit:gnu` is always failing.

Do you have a fix hidden in here? :wink: 
I think I saw this fail on the same test a little earlier, now it isn't failing that test.

**Comment by gschorcht** [comment]
> Could it be that the toolchain on Murdock was upgraded and now `compile/tests/libc_newlib/esp32-wrover-kit:gnu` is always failing?

> Do you have a fix hidden in here? 
> I think I saw this fail on the same test a little earlier, now it isn't failing that test.

I don't know, I saw this problem too. Maybe @kaspar030 updated the toolchain only on one worker and if `tests/libc_newlib` compilation lands on that worker, it will fail :wink:

The problem is not really clear to me, but I can also see it locally. The test complains that the addresses of `printf` and `iprintf` are the same. According to the [comment](https://github.com/32bitmicro/newlib-nano-1.0/blob/f157c994b9a2c4bd8d0cfe9fe8fdd9cd54f8c63b/newlib/README.nano#L32) in test's `README`, these symbols should address the same code in `newlibc-nano`. OK, I forgot to add `newlib_nano` as module. But even if I use
```
USEMODULE=newlib_nano make BOARD=esp32-wroom-32 -C tests/libc_newlib
```
the linker test fails.

**Comment by gschorcht** [comment]
> OK, I forgot to add `newlib_nano` as module when providing the new toolchain. But even if I use
> 
> ```
> USEMODULE=newlib_nano make BOARD=esp32-wroom-32 -C tests/libc_newlib
> ```
> 
> the linker test fails.

Forget it, it't too late today ... With
```
USEMODULE=newlib_nano make BOARD=esp32-wroom-32 -C tests/libc_newlib
```
the test works. The output seems too similar to the failed one.

File: cpu/esp32/periph/rtt.c [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+
+    critical_exit();
+
+    DEBUG("%s rtc_time_saved=%llu rtc_time_diff=%llu "
+          "sys_time_saved=%llu sys_time_offset=%llu\n", __func__,
+          _rtc_time_saved, _rtc_time_diff, _sys_time_saved, _sys_time_offset);
+}
+
+uint32_t _rtt_alarm_saved;
+bool _rtt_alarm_wakeup = false;
```
[the remark this answers is not in the document]
Comment (gschorcht) [reply cpu/esp32/periph/rtt.c]: Indeed.

**Comment by gschorcht** [comment]
I have realized that `%llu` debugging outputs don't work any longer with the `newlib-nano-formatted-io`. According to @kaspar030's comment https://github.com/RIOT-OS/RIOT/pull/13812#issuecomment-613332028 it might be better to replace them by `%u` debugging outputs before this PR is merged instead of reverting `newlib-nano-formated-io`. There are probably other places where `%llu` is used in debugging outputs` and we could replace them by `%u` outputs by a separate PR. What do you think?

**Pull request merged**